src/graph/backend/dnnl/kernels/sdp_primitive.cpp

```diff
@@ -84,7 +84,6 @@ status_t sdp_primitive_kernel_t<quantized>::compile_impl(
         BACKEND_DNNL_ADD_PASS(pipeline, convert_to_runtime_dst_zero_points);
         BACKEND_DNNL_ADD_PASS(pipeline, fuse_dst_zero_points);
     }
-    BACKEND_DNNL_ADD_PASS(pipeline, binary_canonicalization);
```

Comment:
could you please elaborate why this line is deleted?

Reply:
This pass was inherited from large partition kernel where it's used to convert scale input from a scalar value to a 4D value (with shape (1, 1, 1, 1)) as we rely on binary primitive to apply scale and the primitive requires ndims to match for src0 and src1.

It's not needed for sdp_primitive kernel because scale is passed to the primitive directly and no need to convert it to 4D.